### PR TITLE
Fixed syntax error; added null checking where it was lacking

### PR DESCRIPTION
**Pull request by holmesie**
Upon running every Bonnie record against my conversion tool in https://github.com/projecttacoma/cqm-converter, we've discovered some issues with this repo. This PR fixes those issues, and adds additional null checks where they should have been included originally.

---

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: N/A
- [x] JIRA ticket links to this PR N/A
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/datatypes/assessment.js.coffee [hunk 1]
```diff
@@ -111,7 +111,7 @@ class CQL_QDM.AssessmentRecommended extends CQL_QDM.QDMDatatype
     super @entry
     @_authorDatetime = CQL_QDM.Helpers.convertDateTime(@entry.start_time)
     @_method = @entry.method
-    @_negationRationale = @entry.negationReasonv
+    @_negationRationale = @entry.negationReason
```
Comment (losborne) [inline app/assets/javascripts/datatypes/assessment.js.coffee]: I think a unit test for this class with at least the fields set to `null` would be good to add.

File: app/assets/javascripts/datatypes/device.js.coffee [hunk 2]
```diff
@@ -36,7 +36,7 @@ class CQL_QDM.DeviceApplied extends CQL_QDM.QDMDatatype
     if @_anatomicalApproachSite?
       new cql.Code(@_anatomicalApproachSite.code, @_anatomicalApproachSite.code_system)
     else
-      end
+      null
```
Comment (losborne) [inline app/assets/javascripts/datatypes/device.js.coffee]: I think if we mimic the "should show null relevantPeriod" test  in `device_spec.js.coffee`, this line could be covered pretty easily.
